src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -399,21 +405,34 @@ findNeighborRanks( std::vector< vtkBoundingBox > boundingBoxes )
  * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
  * @param[in] comm the MPI communicator
  * @param[in] partitionRefinement number of graph partitioning refinement cycles
+ * @param[in] useGlobalIds if set to one, use global id arrays from the vtk mesh when available. Else, always recompute them.
  */
 vtkSmartPointer< vtkDataSet >
 redistributeMesh( vtkDataSet & loadedMesh,
                   MPI_Comm const comm,
                   VTKMeshGenerator::PartitionMethod const method,
-                  int const partitionRefinement )
+                  int const partitionRefinement,
+                  int const useGlobalIds )
 {
   GEOSX_MARK_FUNCTION;
 
-  // Generate global IDs for vertices and cells
-  vtkNew< vtkGenerateGlobalIds > generator;
-  generator->SetInputDataObject( &loadedMesh );
-  generator->Update();
-  vtkSmartPointer< vtkDataSet > mesh =
-    vtkDataSet::SafeDownCast( generator->GetOutputDataObject( 0 ) );
+  // Generate global IDs for vertices and cells, if needed
+  vtkSmartPointer< vtkDataSet > mesh;
+  if( useGlobalIds != 1
+      || loadedMesh.GetPointData()->GetGlobalIds() == nullptr
+      || loadedMesh.GetCellData()->GetGlobalIds() == nullptr )
```

Comment:
Which means that somebody can request to use provided global ids but something may be "wrong" (`loadedMesh.GetPointData()->GetGlobalIds() == nullptr`) and you end up generating them.

Please change the behavior: if the user asks to use the global ids and it's not possible, please enter an error message and kill the simulation because what's happening is not what is expected and something needs to be fixed (remove the `useGlobalIds` flags or better fill the `GLOBAL_IDS` field...)

Reply:
I fundamentally agree, but in this case, we need a bit more nuance with the useGlobalIds option. If we set it to `true` by default, then the run would abort whenever a mesh doesn't contain ids (and I suspect that's the majority) -- cue a very confused user...
I suggest we allow three values:
 - `useGlobalIds<0`: it will never use global Ids from the mesh, even if they are present
 - `useGlobalIds=0`: the default value, it will use Ids if present and valid, and generate them if not available;
 - `useGlobalIds>0` it will always use mesh Ids, and never generate them. It would abort if they are not available.

what do you think?